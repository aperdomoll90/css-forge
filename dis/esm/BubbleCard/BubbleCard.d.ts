import React from 'react';
import './styles.css';
import '../utils/GlobalStyles.css';
import { BubbleCardPropsType } from './BubbleCard.types';
export declare const BubbleCard: React.FC<BubbleCardPropsType>;
