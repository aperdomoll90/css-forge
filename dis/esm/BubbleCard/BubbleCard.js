var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect } from 'react';
import './styles.css';
import '../utils/GlobalStyles.css';
import { wobbleElement } from '../utils/MouseEvents';
import { LightenDarkenColor } from '../utils/ColorManipulation';
export var BubbleCard = function (_a) {
    var width = _a.width, height = _a.height, primaryColor = _a.primaryColor, secondaryColor = _a.secondaryColor, fontColor = _a.fontColor, label = _a.label, content = _a.content, imgUrl = _a.imgUrl, imgWidth = _a.imgWidth, imgX = _a.imgX, imgY = _a.imgY, hoverX = _a.hoverX, hoverY = _a.hoverY, hoverScale = _a.hoverScale, demoUrl = _a.demoUrl, codeUrl = _a.codeUrl;
    useEffect(function () {
        var elements = document.querySelectorAll('.wobble');
        document.addEventListener('mousemove', function (e) {
            wobbleElement(e, elements);
        });
    }, []);
    var stylesProps = {
        '--width': width ? "".concat(width, "rem") : '17rem',
        '--height': height ? "".concat(height, "rem") : '23rem',
        '--primaryColor': primaryColor ? primaryColor : '#1c2942',
        '--secondaryColor': secondaryColor ? secondaryColor : '#67df6e',
        '--fontColor': fontColor ? fontColor : '#fff',
        '--buttonActive': secondaryColor ? LightenDarkenColor(secondaryColor, -50) : LightenDarkenColor('#67df6e', -50),
        '--imgWidth': imgWidth ? "".concat(imgWidth, "%") : '70%',
        '--imgY': imgY ? "".concat(imgY, "rem") : '3.5rem',
        '--imgX': imgX ? "".concat(imgX, "rem") : '1.8rem',
        '--hoverX': hoverX ? "".concat(hoverX, "rem") : '-3rem',
        '--hoverY': hoverY ? "".concat(hoverY, "rem") : '-5rem',
        '--hoverScale': hoverScale ? hoverScale : 0.8,
    };
    return (_jsx("div", __assign({ className: 'bubbleCard-wrapper', style: stylesProps }, { children: _jsxs("div", __assign({ className: 'bubbleCard-content' }, { children: [_jsx("div", __assign({ className: 'bubbleCard-marquee ' }, { children: _jsx("div", __assign({ "data-speed": '-255', className: 'wobble' }, { children: label })) })), _jsx("div", __assign({ className: 'bubbleCard-img-container' }, { children: _jsx("img", { "data-speed": '-25', alt: "".concat(label, "-image"), src: imgUrl, className: 'bubbleCard-img wobble' }) })), _jsx("h1", __assign({ className: 'bubbleCard-title' }, { children: label })), _jsx("div", __assign({ className: 'bubbleCard-body' }, { children: _jsx("p", __assign({ className: 'bubbleCard-content-text' }, { children: content })) })), _jsxs("div", __assign({ className: 'bubbleCard-footer' }, { children: [demoUrl && (_jsx("a", __assign({ className: 'bubbleCard-link', target: '_blank', href: demoUrl }, { children: _jsx("button", __assign({ className: 'bubbleCard-btn bubbleCard-success' }, { children: "Demo" })) }))), codeUrl && (_jsx("a", __assign({ className: 'bubbleCard-link', target: '_blank', href: codeUrl }, { children: _jsx("button", __assign({ className: 'bubbleCard-btn bubbleCard-border' }, { children: "Github" })) })))] }))] })) })));
};
//# sourceMappingURL=BubbleCard.js.map