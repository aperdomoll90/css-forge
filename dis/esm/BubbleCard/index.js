export { BubbleCard } from "./BubbleCard";
export * from './BubbleCard.types';
//# sourceMappingURL=index.js.map