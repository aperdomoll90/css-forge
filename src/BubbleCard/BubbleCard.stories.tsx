import { Meta, StoryFn } from '@storybook/react'
import { BubbleCard } from './BubbleCard'
import { BubbleCardPropsType } from './BubbleCard.types'
import shoe from '../media/shoe.png'
import dice from '../media/dice.png'
import boat from '../media/boat.png'
import StorybookContainer from '../StorybookContainer'
import { colorPickerControl, percental, mintViolet, pentaDecimals, primaryColors, dozen, negativeArray, scaleValues } from '../utils/StoryProps'
import { blackNwhite } from '../utils/StoryProps'

export default {
  title: 'Cards',
  component: BubbleCard,
  parameters: {
    jest: ['BubbleCard.test.tsx'],
  },
  argTypes: {
    label: { options: ['Sea', 'Game', 'Shoe'], control: { type: 'select' } },
    demoUrl: { options: [null, 'https://ingots.web.app/fishing'], control: { type: 'select' } },
    codeUrl: { options: [null, 'https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll'], control: { type: 'select' } },
    width: { options: pentaDecimals, control: { type: 'select' } },
    height: { options: pentaDecimals, control: { type: 'select' } },
    primaryColor: colorPickerControl,
    secondaryColor: colorPickerControl,
    fontColor: colorPickerControl,
    imgWidth: { options: percental, control: { type: 'select' } },
    imgUrl: { options: [shoe, boat, dice], control: { type: 'select' } },
    imgX: { options: dozen, control: { type: 'select' } },
    imgY: { options: dozen, control: { type: 'select' } },
    hoverX: { options: negativeArray, control: { type: 'select' } },
    hoverY: { options: negativeArray, control: { type: 'select' } },
    hoverScale: { options: scaleValues, control: { type: 'select' } },
    
  },
} as Meta

const BubbleCardTemplate: StoryFn<BubbleCardPropsType> = args => (
  <StorybookContainer>
    <BubbleCard {...args} />
  </StorybookContainer>
)

export const BubbleCardComponent = BubbleCardTemplate.bind({})
BubbleCardComponent.args = {
  label: 'Shoes',
  imgUrl: shoe,
  content: 'Dummy website using shoes as a theme',
  demoUrl: 'https://ingots.web.app/fishing',
  codeUrl: 'https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll',
}
