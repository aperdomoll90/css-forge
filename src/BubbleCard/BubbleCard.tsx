import React, { useEffect, useRef } from 'react'
import './styles.css'
import { wobbleElement } from '../utils/MouseEvents'
import { BubbleCardPropsType } from './BubbleCard.types'

export const BubbleCard : React.FC<BubbleCardPropsType> = ({ color, label, content, imgUrl, demoUrl, codeUrl }): JSX.Element => {
  useEffect(() => {
    const elements = document.querySelectorAll('.wobble')

    document.addEventListener('mousemove', e => {
      wobbleElement(e, elements)
    })
  }, [])

  return (
    <div className='bubbleCard-wrapper'>
      <div className='bubbleCard-content'>
        <div className='bubbleCard-marquee '>
          <div data-speed='-255' className='wobble'>
            {label}
          </div>
        </div>
        <div className='bubbleCard-img-container'>
          <img data-speed='-25' alt={`${label}-image`} src={imgUrl} className='bubbleCard-img wobble' />
        </div>
        <h1 className='bubbleCard-title'>{label}</h1>
        <div className='bubbleCard-body'>
          <p className='bubbleCard-content-text'>{content}</p>
        </div>
        <div className='bubbleCard-footer'>
          <a className='bubbleCard-link' target='_blank' href={demoUrl}>
            <button className='bubbleCard-btn bubbleCard-success'>Demo</button>
          </a>
          <a className='bubbleCard-link' target='_blank' href={codeUrl}>
            <button className='bubbleCard-btn bubbleCard-border'>Github</button>
          </a>
        </div>
      </div>
    </div>
  )
}


