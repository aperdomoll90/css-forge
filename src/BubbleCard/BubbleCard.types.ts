export interface BubbleCardPropsType {
  color?: string
  label: string
  content: string
  imgUrl: string
  demoUrl: string
  codeUrl: string
}
