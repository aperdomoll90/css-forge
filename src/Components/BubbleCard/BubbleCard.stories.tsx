import { Meta, StoryFn } from '@storybook/react'
import BubbleCard, { BubbleCardPropsType } from './BubbleCard'
import StorybookContainer from '../StorybookContainer'
import shoe from './media/shoe.png'


export default {
  title: 'Cards',
  component: BubbleCard,
  parameters: {
    jest: ['BubbleCard.test.tsx'],
  },
  argTypes: {
    color: { options: ['red', ''], control: { type: 'radio' } },
    label: { options: ['Sea', 'Forest', 'Mountain',], control: { type: 'radio' } },
  },
} as Meta

const BubbleCardTemplate: StoryFn<BubbleCardPropsType> = (args) => (
  <StorybookContainer>
    <BubbleCard {...args} />
  </StorybookContainer>
)

export const BubbleCardComponent = BubbleCardTemplate.bind({})
BubbleCardComponent.args = {
  label: 'Shoes',
  imgUrl: shoe,
  content: 'Dummy website using shoes as a theme',
  demoUrl: 'https://ingots.web.app/fishing',
  codeUrl: 'https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll',
}