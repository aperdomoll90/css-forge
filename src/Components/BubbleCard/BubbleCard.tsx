import { useEffect, useRef } from 'react'
import './styles.css'

export interface BubbleCardPropsType {
  color?: string
  label: string
  content: string
  imgUrl: string
  demoUrl: string
  codeUrl: string
}

function BubbleCard({ color, label, content, imgUrl, demoUrl, codeUrl }: BubbleCardPropsType) {
  const areaRef = useRef()

  const handleMouseMov = e => {
    const width = window.innerWidth
    const height = window.innerHeight
    // get mouse position
    const x = e.clientX
    const y = e.clientY
    // find the middle
    const middleX = window.innerWidth / 2
    const middleY = window.innerHeight / 2

    //get offset from middle
    const offsetX = ((x - middleX) / middleX) * 85
    const offsetY = ((y - middleY) / middleY) * 85

    const wobbles = document.querySelectorAll('.wobble')

    wobbles.forEach(wobble => {
      const speed = wobble.getAttribute('data-speed')
      const x = (offsetX / width) * speed
      const y = (offsetY / height) * speed
      wobble.style.transform = `translateX(${x}%) translateY(${y}%) `
    })
  }

  return (
    <div className='bubbleCard-wrapper'>
      <div className='bubbleCard-content' ref={areaRef as any} onMouseMove={handleMouseMov}>
        <div className='bubbleCard-marquee '>
          <div data-speed='-255' className='wobble'>
            {label}
          </div>
        </div>
        <div className='bubbleCard-img-container'>
          <img data-speed='-25' alt={`${label}-image`} src={imgUrl} className='bubbleCard-img wobble' />
        </div>
        <h1 className='bubbleCard-title'>{label}</h1>
        <div className='bubbleCard-body'>
          <p className='bubbleCard-content-text'>{content}</p>
        </div>
        <div className='bubbleCard-footer'>
          <a className='bubbleCard-link' target='_blank' href={demoUrl}>
            <button className='bubbleCard-btn bubbleCard-success'>Demo</button>
          </a>
          <a className='bubbleCard-link' target='_blank' href={codeUrl}>
            <button className='bubbleCard-btn bubbleCard-border'>Github</button>
          </a>
        </div>
      </div>
    </div>
  )
}

export default BubbleCard
