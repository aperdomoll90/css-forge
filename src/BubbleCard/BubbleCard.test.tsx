import { render } from '@testing-library/react'
import { BubbleCard } from './BubbleCard'
import shoe from './media/shoe.png'
import { BubbleCardPropsType } from './BubbleCard.types'

describe('BubbleCard component', () => {
  // const { axe, toHaveNoViolations } = require('jest-axe')
  // expect.extend(toHaveNoViolations)
  const props: BubbleCardPropsType = {
    label: 'label',
    imgUrl: shoe,
    content: 'This card follows the pointer',
    demoUrl: 'https://ingots.web.app/fishing',
    codeUrl: 'https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll',
  }

  const renderComponent = () => render(<BubbleCard {...props} />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
  //   it('should demonstrate ADA compliance on this component', async () => {
  //     const { container } = renderComponent()
  //     const results = await axe(container)
  //     expect(container).toHaveNoViolations()
  //   })
})
