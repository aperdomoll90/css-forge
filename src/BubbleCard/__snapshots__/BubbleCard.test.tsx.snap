// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BubbleCard component should match snapshots 1`] = `
<div>
  <div
    class="bubbleCard-wrapper"
  >
    <div
      class="bubbleCard-content"
    >
      <div
        class="bubbleCard-marquee "
      >
        <div
          class="wobble"
          data-speed="-255"
        >
          label
        </div>
      </div>
      <div
        class="bubbleCard-img-container"
      >
        <img
          alt="label-image"
          class="bubbleCard-img wobble"
          data-speed="-25"
          src="shoe.png"
        />
      </div>
      <h1
        class="bubbleCard-title"
      >
        label
      </h1>
      <div
        class="bubbleCard-body"
      >
        <p
          class="bubbleCard-content-text"
        >
          This card follows the pointer
        </p>
      </div>
      <div
        class="bubbleCard-footer"
      >
        <a
          class="bubbleCard-link"
          href="https://ingots.web.app/fishing"
          target="_blank"
        >
          <button
            class="bubbleCard-btn bubbleCard-success"
          >
            Demo
          </button>
        </a>
        <a
          class="bubbleCard-link"
          href="https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll"
          target="_blank"
        >
          <button
            class="bubbleCard-btn bubbleCard-border"
          >
            Github
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
